app/controllers/api/order_cycles_controller.rb

```diff
@@ -45,7 +45,7 @@ def properties
     private
 
     def render_no_products
-      render status: :not_found, json: ''
+      render status: :not_found, plain: {}.to_json, content_type: "application/json"
```

Comment:
I think we can just do `json: {}` here, instead of `plain: {}.to_json`.

Also, explicitly declaring the content type in every response doesn't look right. I think in other API controllers we have something that sets the content type globally... I think we should figure out why this controller isn't doing it and handle it that way instead (and remove all these explicit content type declarations).

Reply:
It looks like we usually set it in `Api::BaseController` [here](https://github.com/openfoodfoundation/openfoodnetwork/blob/ec4d171993f9a5b2e1ac48a517f8c0806deed87d/app/controllers/api/base_controller.rb#L68).

In the case of the Angular Resource that's returning these products ([here](https://github.com/openfoodfoundation/openfoodnetwork/blob/ec4d171993f9a5b2e1ac48a517f8c0806deed87d/app/assets/javascripts/darkswarm/services/order_cycle_resource.js.coffee#L6)), it isn't declaring the format (other Resources seem to do this correctly, example [here](https://github.com/openfoodfoundation/openfoodnetwork/blob/ec4d171993f9a5b2e1ac48a517f8c0806deed87d/app/assets/javascripts/darkswarm/services/shops_resource.js.coffee#L6)), and I think that's what is causing the `content_type` of the response to not be set.

Reply:
We could potentially also set the content_type to _default_ to `application/json` for API actions if no format is given..?

Reply:
There's some API cleanup / tech-debt highlighted here as well, maybe for another issue. In the `Api::BaseController` we're checking for XML format in the request, but I'm pretty sure none of our API controllers render XML under any circumstances, so that looks like dead code to me. And we should probably be setting the content type to `application/json` on all API response headers unconditionally, because that's what they return...

Reply:
I played around with some different things; `json: {}` did not seem to work for some reason. Then I noticed that we do `plain:` and specify the content type on line 23 in that file and... I took the lazy way out :) 

I like your approach in https://github.com/openfoodfoundation/openfoodnetwork/pull/6578 much better, and it looks like the specs are passing except for some flakiness. Should we close this PR and move discussion to #6578 ?

Reply:
Sure :+1: